tools/generate_schema_wrapper.py

```diff
@@ -894,8 +896,13 @@ def generate_encoding_artifacts(
         if info.supports_arrays:
             docstring_union_types.append("List")
             tp_inner = f"OneOrSeq[{tp_inner}]"
-        signature_args.append(f"{channel}: Optional[{tp_inner}] = Undefined")
-        typed_dict_args.append(f"{channel}: {tp_inner}")
+
+        alias_name = f"Channel{channel[0].upper()}{channel[1:]}"
+        channel_type_aliases.append(f"{alias_name}: TypeAlias = {tp_inner}")
```

Comment:
@binste you can't reuse this outside of annotation scope until `py3.10` IIRC, due to ` | `

Reply:
Ignore me if `ruff` has already given you the warning

Reply:
Wow, live error feedback, thanks! :) I'll need to get used to the fact that we can now taste some of the improved typing syntax of newer versions thanks to the futures import but not in all scopes.

Reply:
Actually, ruff did not give me any warning although I have the ruff vscode extension installed. Any setting you can recommend for this?

Reply:
> Actually, ruff did not give me any warning although I have the ruff vscode extension installed. Any setting you can recommend for this?

You can run `hatch` commands against an earlier python version for this

https://github.com/vega/altair/blob/main/CONTRIBUTING.md#failures-on-specific-python-versions

I added it to the guide, but maybe only @mattijn saw

Reply:
There are some other changes on there as well, you should be able to run everything from `hatch` now

Reply:
Great, thanks!